upf/simplifier.py

```diff
@@ -154,6 +154,12 @@ def walk_implies(self, expression: FNode, args: List[FNode]) -> FNode:
         else:
             return self.manager.Implies(sl, sr)
 
+    def walk_exists(self, expression: FNode, args: List[FNode]) -> FNode:
+        return self.manager.Exists(args[0], *expression.variables())
```

Comment:
Maybe handle the case where `args[0]` is true or false. Moreover, we can simplify the cases where `args[0]` does not contain some of the variables in `expression.variables()` by reducing the resulting set of variables.

E.g.
`Exists a, b, c . Or(b, c)` is equivalent to `Exists b, c . Or(b, c) `

Reply:
How? The first solution that comes to my mind is that args is a:
List[Tuple[FNode, Set[Variable]]] and the function returns a Tuple[FNode, Set[Variable]]. What do you think about this?